### PR TITLE
Dev 3747 new exercise about handling different http request methods

### PR DESCRIPTION
**Pull request by eslopfer**

### CODE REVIEW

File: subjects/uninvited/README.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+- It has to listen on port `5000` and it will have to print a simple message on the console, specifying the listening port.
+- Its HTTP response should contain a coherent status code depending on the handling of the received HTTP request. More specifically, your server should be able to respond with the following status codes: `201` and `500`.
+- The responses will always be JSON and this information should be explicit in the HTTP response.
+- For each http `POST` request, your program should create the corresponding JSON file and store the contents of the body, and then provide the content as JSON in the HTTP response, if possible. If the file already exists, it should replace it.
+- If for any reason the server fails, the response should be an object with a key `error` and its value `server failed`.
+
+### Example
+
+To test your program, you should be able to expect the following behaviour once your program is up and running.
```
Comment (mikysett) [inline subjects/uninvited/README.md:23]: small typo on "behavior"

**Review by mikysett — CHANGES_REQUESTED** [review]
In order to check the tests may you add the solution in the `piscine-js-solutions` repo please

File: js/tests/test.mjs [hunk 2]
```diff
@@ -5,9 +5,9 @@ import { deepStrictEqual } from 'assert'
 import { fileURLToPath } from 'url'
 import { tmpdir } from 'os'
 import http from 'http'
-import fs from 'fs'
 
 global.window = global
+global._fetch = fetch
```
[the remark this answers is not in the document]
Comment (nprimo) [reply js/tests/test.mjs:11]: This change has already been implemented and merged in the current master version - I guess having it in this PR should not impact the final `master` version

File: subjects/uninvited/README.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  "drink": "alcohol",
+  "food": "bats"
+}
+```
+
+### Notions
+
+- [HTTP protocol](https://developer.mozilla.org/en-US/docs/Web/HTTP)
+- [HTTP Status Codes](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status)
+- [Node http package: createServer](https://nodejs.org/en/knowledge/HTTP/servers/how-to-create-a-HTTP-server/)
```
Comment (nprimo) [inline subjects/uninvited/README.md:38]: I would add this link as well on how to read data from a POST request 
https://nodejs.org/en/knowledge/HTTP/servers/how-to-read-POST-data/

File: js/tests/uninvited_test.mjs [hunk 4]
```diff
@@ -0,0 +1,198 @@
+import { once } from 'node:events'
+import { spawn } from 'node:child_process'
+import { mkdir, writeFile, chmod } from 'fs/promises'
+import { join } from 'path'
+import fs from 'node:fs/promises'
+import fss from 'fs'
```
Comment (nprimo) [inline js/tests/uninvited_test.mjs:6]: This is imported but not used

**Review by nprimo — CHANGES_REQUESTED** [review]
I have added a few inline comments. 

Moreover, all the changes made to `test.mjs` are already implemented in the current `master` repo version - I don't know if it make sense to include these changes in this PR as well. Overall it should not influence the final result of `master` after this PR is merged

**Review by nprimo — APPROVED** [review]
LGTM!

File: .gitignore [hunk 5]
```diff
@@ -1,3 +1,3 @@
 node_modules
 rust/**/target/
-js/soluss
+js/soluss
```
Comment (mikysett) [inline .gitignore:3]: here a newline is missing at the end of the file, maybe we could put it back, this also will prevent for seeing a change in the `.gitignore` while actually nothing changed.

File: subjects/uninvited/README.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+### Example
+
+To test your program, you should be able to expect the following behaviour once your program is up and running.
+
+```shell
+curl -X POST localhost:5000/Ozzi_Osbourne -H "Content-Type: application/json" -d '{"answer": "yes", "drink": "dark", "food": "bats"}'
+{
+  "answer": "yes",
+  "drink": "dark",
+  "food": "bats"
```
Comment (mikysett) [inline subjects/uninvited/README.md:30]: maybe it's worth changing also `bats` to something more neutral.

File: subjects/uninvited/README.md (hunk 1 above)
Comment (mikysett) [reply subjects/uninvited/README.md:23]: actually it's not a typo, just UK English... I've checked in the curriculum, there is both versions, US `behavior` and UK `behaviour`, in our subjects.
I have to say we have more times the US version than the UK, I suggest we use that one, but I have no strong opinion so feel free to chose the path you prefer about it.

**Review by mikysett — CHANGES_REQUESTED** [review]

**Review by mikysett — APPROVED** [review]
LGTM

**Pull request merged**